pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -144,15 +150,37 @@ func (r *ReconcileConstraintTemplate) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
+	dirty := false
```

Comment:
I suggest you make the "dirty" bit be a property of the registry itself and have report() return early if it is not true.

Reply:
+1